pkgs/by-name/mi/midisheetmusic/package.nix

```diff
@@ -3,26 +3,44 @@
   stdenv,
   fetchurl,
   mono,
-  dotnetPackages,
+  unzip,
+  runCommand,
+  mkNugetDeps,
   makeWrapper,
+  makeFontsConf,
   gtk2,
   cups,
   timidity,
 }:
 
 let
-  version = "2.6";
+  deps = mkNugetDeps {
+    name = "midisheetmusic-deps";
+    nugetDeps =
+      { fetchNuGet }:
+      [
+        (fetchNuGet {
+          pname = "NUnit.Console";
+          version = "3.0.1";
+          sha256 = "154bqwm2n95syv8nwd67qh8qsv0b0h5zap60sk64z3kd3a9ffi5p";
+        })
+        (fetchNuGet {
+          pname = "NUnit";
+          version = "2.6.4";
+          sha256 = "1acwsm7p93b1hzfb83ia33145x0w6fvdsfjm9xflsisljxpdx35y";
+        })
+      ];
+  };
 in
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "midisheetmusic";
-  inherit version;
+  version = "2.6";
 
   src = fetchurl {
-    url = "mirror://sourceforge/midisheetmusic/MidiSheetMusic-${version}-linux-src.tar.gz";
+    url = "mirror://sourceforge/midisheetmusic/MidiSheetMusic-${finalAttrs.version}-linux-src.tar.gz";
```

Comment:
I've seen this pattern discouraged in other reviews because it doesn't provide a way to override the hash.  Thus changing the version will never actually work (unless you override `src`).

Reply:
so if I understand correctly, it'd be better if the version number was hardcoded in the url?

Reply:
Or `let` or `rec` I suppose.  @GGG-KILLER I think you were the one who brought this up the last time I saw it on a review, but I don't recall what package it was for.  What do you think?

I think in this case it's not much of a practical problem, because changing the version will change `src.name`.  With other fetchers (or if the url was like `${finalAttrs.version}/src.tar.gz`, you can run into the problem where `src` still ends up pointing to the old version.

Reply:
I don't remember whoever brought it up in a PR of mine but it is advice I've been following ever since.

Using `finalAttrs.version` instead of `rec`/`let in` is advised against because it introduces an "`.override` trap" where the user overrides the version and thinks things changed because `src.url` has changed, however because `src.hash` is the same, nothing new is downloaded (which I'd actually consider a bug on its own but that's a whole different can of worms). This issue seems to have been first brought up in #310373.

However, standardization is pending on #315337, so I believe blocking PRs based on this is unnecessary as there's no official guidance nor consensus on it. So I'd leave it up to @mdarocha whether they think it's significant enough to change or not.

Reply:
Alright, I've updated it to use `let ... in` instead.